packages/docusaurus-plugin-client-redirects/src/collectRedirects.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {flatten, uniqBy} from 'lodash';
+import {
+  RedirectsCreator,
+  PluginContext,
+  RedirectMetadata,
+  PluginOptions,
+} from './types';
+import {
+  fromExtensionsRedirectCreator,
+  toExtensionsRedirectCreator,
+} from './redirectCreators';
+
+export default function collectRedirects(
+  pluginContext: PluginContext,
+): RedirectMetadata[] {
+  const redirects = doCollectRedirects(pluginContext);
+  return filterUnwantedRedirects(redirects, pluginContext);
+}
+
+function filterUnwantedRedirects(
+  redirects: RedirectMetadata[],
+  pluginContext: PluginContext,
+): RedirectMetadata[] {
+  // TODO how should we warn the user of filtered redirects?
+
+  // we don't want to create twice the same redirect
+  redirects = uniqBy(redirects, (redirect) => redirect.fromRoutePath);
+
+  // We don't want to override an existing route
+  redirects = redirects.filter(
+    (redirect) => !pluginContext.routesPaths.includes(redirect.fromRoutePath),
+  );
+
+  return redirects;
```

Comment:
@lex111 @yangshun is there a decent way to warn the user in case conflicting redirects are found?

For example when we detect one of his redirect rules would override an existing path, or if he ends up having twice the same redirect or something?

Nothing prevents the user to do weird things like `createRedirects(originalPath) => [originalPath + ".html",originalPath + ".html"]`

For now I just silently "filter" those to avoid overriting existing route file, or writing twice the same redirect, but maybe it should warn the user instead?